### ISSUE TITLE
missing translation of Nutrition Information

### ISSUE THREAD

**Issue by cloud2018** [nextcloud/cookbook#495]
**Description**
1. In edit mode in block Nutrition Information "Select option" is not translated and the string is not available in Transifex
1. In view mode in block Nutrition Information the names of nutrions are not translated, but the strings are available in Transifex and are used in the drop down box

**Screenshot 1**
![cb1](https://user-images.githubusercontent.com/34692335/104129071-476f1380-536b-11eb-84c4-a554edb1da23.png)

**Screenshot 2**
![cb2](https://user-images.githubusercontent.com/34692335/104129072-4b029a80-536b-11eb-93e2-842393e5aee4.png)

**Comment by christianlupus** [comment]
In general this should be translatable. In the [vue files](https://github.com/nextcloud/cookbook/blob/493d21e0608cd4d089122676e766927c3cd0fc26/src/components/RecipeEdit.vue#L79-L90) the translation is prepared. This is most probably an issue with the parser of transifex.

**Comment by cloud2018** [comment]
@christianlupus yes, when editing the nutrions are translated ("Zuckergehalt" in 1st screenshot) as in  your linked file cookbook/src/components/RecipeEdit.vue. Only "Select option" is not translated, perhaps a nextcloud issue, not a cookbook issue.

But in the view-mode the nutrions are not translated. Maybe because the t-functions inside the titles are surrounded by ",  https://github.com/nextcloud/cookbook/blob/493d21e0608cd4d089122676e766927c3cd0fc26/src/components/RecipeView.vue#L62

**Comment by christianlupus** [comment]
@cloud2018 No I doubt that. The `"` are there from the Vue.js requirements. The must be there. See e.g. [here](https://github.com/nextcloud/cookbook/blob/5e708abbba8a0322ee3b7cc6c60dbd3fcc5a8061/src/components/AppNavi.vue#L25) where it is working.

I just checked and found that the translations seem to be exported from the vue files correctly but are somehow not transferred to transifex. I opened an issue there and now we have to see what's happening next.

**Comment by christianlupus** [comment]
I just found a workaround for the problem but that needs a complete overhaul of the code base. In fact, after the translation no more arguments should be given. Exampe see [the upstream bug report](https://github.com/nextcloud/docker-ci/issues/259#issuecomment-778321169).

**Comment by seyfeb** [comment]
So you’re saying that writing 

```
<recipe-nutrition-info-item v-if="('sugarContent' in recipe.nutrition) && !isNullOrEmpty(recipe.nutrition['sugarContent'])" :title="t('cookbook', 'Sugar')" :data="recipe.nutrition['sugarContent']" />
```

as

```
<recipe-nutrition-info-item v-if="('sugarContent' in recipe.nutrition) && !isNullOrEmpty(recipe.nutrition['sugarContent'])"
  :title="t('cookbook', 'Sugar')" 
  :data="recipe.nutrition['sugarContent']" />
```

should solve the issue as long as the upstream bug is present? Then, I would update those to have a quick fix.

**Comment by christianlupus** [comment]
Yes, this is correct. In the example above the regex currently in use by the transifex integration will use `Sugar')" :data="recipe.nutrition['sugarContent` as the original (English) text which in turn will break other file formats.

One remark: I am just working on issue #532 and #531. During the run of prettier (part of it), the suggested changes from you will be auto-generated. That would speed up that refactoring process much.

**Comment by seyfeb** [comment]
Alright, I leave you to it then!

**Comment by cloud2018** [comment]
Is there a workaround for untranslated "Select option" from screenhot 1, too?

**Comment by seyfeb** [comment]
It wouldn’t’ve worked for “Select option” but should now.

**Comment by christianlupus** [comment]
With #607 the missing translations should be available by the next nightly run of the nextcloud bot.

I will close here as we have to wait for the next release to get the translations into the app.
